### ISSUE TITLE
parser: latex support?

### ISSUE THREAD

**Issue by v2gt** [orgapp/orgajs#41]
希望:
1. parser解析出latex公式: $a+b$, \[a+b \]
2. orga-rehype 生成的html直接就可以展示数学公式，比如用katex

**输入org:**
```
* test

  $a+b$


  \[a+b+\alpha\]
```

**html**
`<div class="section"><h1>test</h1><p>$a+b$</p><p>\[a+b+\alpha\]</p></div>`

> 会的，这个 parser 会尽量跟org-mode syntax 12保持一致。对于parser本身，我觉得最make sense 的实现是直接返回node:
> 
> {
>   type: “latex”,
>   name: “equation”,
>   value: “x=\sqrt{b}”
> }
> 具体的render可以在org-mode to HTML 层用其他包来实现。这样其实就没有多少工作在parser这边。

**Comment by gitonthescene** [comment]
In #49 I suggest adding plugin support.  With that, [one of these](https://unifiedjs.com/explore/keyword/latex/) might be helpful.

**Comment by LaloHao** [comment]
Here's a POC of using LaTeX

Live: https://blorg-latex.hao.systems/using-latex
Source: https://github.com/LaloHao/gatsby-theme-blorg-latex

**Comment by gitonthescene** [comment]
> Here's a POC of using LaTeX
> 
> Live: https://blorg-latex.hao.systems/using-latex
> Source: https://github.com/LaloHao/gatsby-theme-blorg-latex

@LaloHao Is this all done with the katex import?  I.e. did this require zero changes to orgajs?  I’m you didn’t have to render the latex to the AST at all right?

**Comment by LaloHao** [comment]
> > Here's a POC of using LaTeX
> > Live: https://blorg-latex.hao.systems/using-latex
> > Source: https://github.com/LaloHao/gatsby-theme-blorg-latex
> 
> @LaloHao Is this all done with the katex import? I.e. did this require zero changes to orgajs? I’m you didn’t have to render the latex to the AST at all right?

Yes, you don't need to change orgajs internals, you can do it on your own project

There are 2 commits on my example repository:
1. First one is the original source code
2. [Second one is the changes to you add to use katex](https://github.com/LaloHao/gatsby-theme-blorg-latex/commit/2ad15d7b5bc5cdccda1cb34d8f74b61b40d33974)

How to:
1. I ran `npm install --save katex rehype-katex rehype-math rehype-parse` for the dependencies

2. I added the css file to `gatsby-browser.js`
```js
require('katex/dist/katex.min.css');
```

3. [Create a new component post with latex processing](https://github.com/LaloHao/gatsby-theme-blorg-latex/blob/2ad15d7b5bc5cdccda1cb34d8f74b61b40d33974/src/gatsby-theme-blorg/components/post.tsx#L9)

4. [Create org-mode posts](https://github.com/LaloHao/gatsby-theme-blorg-latex/blob/2ad15d7b5bc5cdccda1cb34d8f74b61b40d33974/content/using-latex.org)
